src/transformers/utils/hub.py

```diff
@@ -253,6 +253,20 @@ def _patch_hf_hub_tqdm():
     huggingface_hub.file_download.tqdm = old_tqdm
 
 
+def get_redirected_link(repo_id):
+    r"""
+    Tries to get a redirected link, otherwise return the same repo_id. Code inspired from:
+    https://stackoverflow.com/questions/20475552/python-requests-library-redirect-new-url
+    """
+    temp_link = f"{HUGGINGFACE_CO_RESOLVE_ENDPOINT}/{repo_id}"
+    try:
+        r = requests.head(temp_link, allow_redirects=True)
+        repo_id = r.url.replace(f"{HUGGINGFACE_CO_RESOLVE_ENDPOINT}/", "")
+    except BaseException:
+        pass  # Do nothing if the link is not redirected
+    return repo_id
```

Comment:
Doesn't seem linked to this PR.

Reply:
Yeah a bad rebase happened, I am trying to fix it now

Reply:
Now it should be completely removed from the git history